SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -2443,6 +2443,53 @@ void USpatialNetDriver::RefreshActorDormancy(AActor* Actor, bool bMakeDormant)
 	}
 }
 
+void USpatialNetDriver::RefreshActorVisibility(AActor* Actor, bool bMakeVisible)
+{
+	check(IsServer());
+	check(Actor);
+
+	const Worker_EntityId EntityId = PackageMap->GetEntityIdFromObject(Actor);
+	if (EntityId == SpatialConstants::INVALID_ENTITY_ID)
+	{
+		UE_LOG(LogSpatialOSNetDriver, Verbose, TEXT("Unable to change visibility on an actor without entity id. Actor's Name: %s"), *Actor->GetName());
+		return;
+	}
+
+
+	const bool bHasAuthority = StaticComponentView->HasAuthority(EntityId, SpatialConstants::VISIBLE_COMPONENT_ID);
+	if (bHasAuthority == false)
+	{
+		UE_LOG(LogSpatialOSNetDriver, Log, TEXT("Unable to  change visibility on an actor without authority. Actor's Name: %s "), *Actor->GetName());
+		return;
+	}
+
+	const bool bVisibilityComponentExists = StaticComponentView->HasComponent(EntityId, SpatialConstants::VISIBLE_COMPONENT_ID);
+
+	// If the Actor is Visible make sure it has the Visible component
+	if (bMakeVisible)
```

Comment:
I guess the if statements here could be collapsed to
`if (bMakeVisible && !bVisibilityComponentExists)`
`else if (!bMakeVisible && bVisibilityComponentExists)`
but that's just my preference.